FV3/atmos_cubed_sphere/tools/fv_io.F90

```diff
@@ -1016,4 +1041,188 @@ subroutine fv_io_read_BCs(Atm)
     return
   end subroutine fv_io_read_BCs
 
+!>@brief The subroutine 'cubed_a2d' transforms the wind from the A Grid to the D Grid.
+! Vendored from external_ic.F90 to overcome a circular dependency.
```

Comment:
It was awkward that I needed to copy this over from `external_ic.F90`.  I'm not sure if there is another way around this unfortunately.

Reply:
good point. usually you want to consolidate, but it's difficult to weight that against making a small impact on the model to gain acceptance.

Reply:
It does seem like a nice general purpose routine (tbh I'm surprised it's not used to transform the wind tendencies from the physics). Where would be a good place for the subroutine to go? Maybe `fv_grid_tools.F90` although a lot of the routines in there are more to do with grid/mesh generation. Would be useful to get an opinion from a DSL person here.

Reply:
If we were to move it, I agree `fv_grid_tools.F90` would probably be the place.  That's where the reverse operation, `cubed_to_latlon`, currently lives: https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/361a7c7c9b86284b9064ecec09e6bb4a37a94bbc/FV3/atmos_cubed_sphere/model/fv_grid_utils.F90#L2345-L2363

Reply:
Opened #36 to track this going forward.